cmake/kokkos_compiler_id.cmake

```
@@ -47,6 +48,11 @@ IF(Kokkos_ENABLE_CUDA)
     SET(INTERNAL_USE_COMPILER_LAUNCHER true)
   ENDIF()
 ENDIF()
+ELSE()
+  IF(Kokkos_ENABLE_CUDA)
+    kokkos_internal_have_compiler_nvcc(${CMAKE_CUDA_COMPILER})
+  ENDIF()
+ENDIF()
```

```suggestion
ELSEIF(Kokkos_ENABLE_CUDA)
  kokkos_internal_have_compiler_nvcc(${CMAKE_CUDA_COMPILER})
ENDIF()
```
Alternatively, I would prefer having
```
IF(Kokkos_ENABLE_CUDA)
```
as the outer condition.

---

done